File: matlab/rifta/zero_pad.m
function F = zero_pad(F0,mm,nn)
%expand f0 to [m n]
%this function can be realized by padarray, but is slower

%size f0
[m, n]=size(F0); 

%generate a larger matrix with size [mm nn]
F=zeros(mm,nn);

%copy original data
F(1:m,1:n)=F0;

end

File: python/rifta/zero_pad.py
import numpy as np

def zero_pad(F0,mm,nn):
    'size f0'
    m, n = F0.shape
    '生成一个更大的矩阵，大小为[mm nn]'
    F = np.zeros((mm, nn))
    '复制原始数据'
    F[:m, :n] = F0
    return F